### PR TITLE
Refactor config manager

### PR DESCRIPTION
**Pull request by erwint**
Based on #171 cleaned up the projects configurators, so that applying natures is done in the same way in all of them.

### CODE REVIEW

File: com.basistech.m2e.code.quality.checkstyle.test/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -7,4 +7,5 @@ Bundle-Vendor: Basis Technology Corp.
 Fragment-Host: com.basistech.m2e.code.quality.checkstyle;bundle-version="[1.1.0,2.0.0)"
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7
 Require-Bundle: org.junit,
+ com.basistech.m2e.code.quality.shared.test;bundle-version="1.1.0",
```
Comment (vorburger) [inline com.basistech.m2e.code.quality.checkstyle.test/META-INF/MANIFEST.MF]: Perhaps the Version number here (and elsewhere like it?) can just be dropped?

```suggestion
 com.basistech.m2e.code.quality.shared.test,
```

**Comment by vorburger** [comment]
@erwint will you rebase and resolve conflicts?

File: com.basistech.m2e.code.quality.checkstyle.test/META-INF/MANIFEST.MF (hunk 1 above)
Comment (erwint) [reply com.basistech.m2e.code.quality.checkstyle.test/META-INF/MANIFEST.MF]: changed as requested in all 4 places

File: com.basistech.m2e.code.quality.checkstyle/src/main/java/com/basistech/m2e/code/quality/checkstyle/EclipseCheckstyleProjectConfigurator.java [hunk 2]
```diff
@@ -102,16 +101,22 @@ protected void handleProjectConfigurationChange(
...
-					        mavenCheckstyleConfig);
-					addNature(project, CheckstyleNature.NATURE_ID, monitor);
-				} else {
-					deleteEclipseFiles(project, monitor);
-					removeNature(project, CheckstyleNature.NATURE_ID, monitor);
+					buildCheckstyleConfiguration(pcWorkingCopy, mavenCheckstyleConfig);
+					configure(project, false, monitor);
+					// found a non-skipped config
+					applied = true;
+					break;
```
Comment (vorburger) [inline com.basistech.m2e.code.quality.checkstyle/src/main/java/com/basistech/m2e/code/quality/checkstyle/EclipseCheckstyleProjectConfigurator.java]: NOK, see #180 and revert in #182

**Pull request merged**